qlib/HttpServer.qm

```diff
@@ -1118,6 +1149,30 @@ public class HttpServer::HttpServer inherits HttpServer::AbstractLogger {
         listeners{id}.addHandlers(h);
     }
 
+    #! remove request handler from a listener given the listener's id
+    removeHandlerFromListenerID(softstring id, string handler_name) {
+        lm.enter();
+        on_exit lm.exit();
+
+        if (!listeners.hasKey(id)) {
+            throw "HTTP-SERVER-ERROR", sprintf("there is no listener with ID %d", id);
```

Comment:
I think `INVALID-LISTENER` would be better

Reply:
Other code relies on this error being `HTTP-SERVER-ERROR`, therefore for backwards compatibility I wouldn't change this.